daintree/run_fit_models.py

```diff
@@ -170,25 +170,59 @@ def gather_ensemble_tasks(
         cors = (
             pd.DataFrame(cors)
             .reset_index()
-            .rename(columns={"index": "gene", 0: "pearson"})
+            .rename(columns={"index": "target_variable", 0: "pearson"})
         )
         cors["model"] = model
 
         all_cors.append(cors)
 
     all_cors = pd.concat(all_cors, ignore_index=True)
-    ensemble = all_features.merge(all_cors, on=["gene", "model"])
+    ensemble = all_features.merge(all_cors, on=["target_variable", "model"])
+    
+    ### The following code is implemented this way due to a
+    # PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.
+    # De-fragment the DataFrame
+    ensemble = ensemble.copy()
 
-    # Get the highest correlation across models per "gene" (entity)
-    ensemble["best"] = ensemble.groupby("gene")["pearson"].rank(ascending=False) == 1
+    # Get the highest correlation across models per "target_variable" (entity)
+    ranked_pearson = ensemble.groupby("target_variable")["pearson"].rank(ascending=False)
+    ensemble["best"] = (ranked_pearson == 1)
 
-    ensb_cols = ["gene", "model", "pearson", "best"]
+    ensb_cols = ["target_variable", "model", "pearson", "best"]
 
-    for i in range(top_n):
-        cols = ["feature" + str(i), "feature" + str(i) + "_importance"]
-        ensb_cols += cols
+    # Iterate over each row in the ensemble
+    for index, row in ensemble.iterrows():
+        # Get the target variable name from the ensemble
+        target_variable = row['target_variable']
+        
+        # Extract the corresponding target data from `targets` DataFrame
+        y = targets[target_variable]
+
+        # Iterate over feature columns in the ensemble
+        for i in range(top_n):  # Assuming there are 50 features (feature0 to feature49)
+            feature_col = f'feature{i}'
+            feature_name = row[feature_col]  # Get the feature name listed in the ensemble row
+            
+            if feature_name in features.columns:
+                # Extract feature data from the features DataFrame
+                x = features[feature_name]
+                
+                # Compute Pearson correlation, handling constant columns
+                if x.std() == 0 or y.std() == 0:
+                    corr = None  # Assign None or NaN if feature or target is constant
+                else:
+                    corr, _ = pearsonr(x, y)
+                
+                # Add correlation as a new column to the ensemble
+                ensemble.loc[index, f'{feature_col}_correlation'] = corr
```

Comment:
I think this works, but I'll offer the more canonical way do something like this would be to construct the contents of the entire column and then add it to the dataframe as a new column.

for example, instead of first iterating over rows, like:
```
for index, row in ensemble.iterrows():
   ...
   for i in range(top_n):
      ...
```

You could flip this to be:
```
for i in range(top_n):
   ...
   for index, row in ensemble.iterrows():
      ...
```

Now, you have an inner loop which is processing all the rows in `ensemble` to create a new column `f'{feature_col}_correlation']`

Now for readability, I'd lift the calculation that is performed per row into it's own function. That'd look something like:

```
def calc_feature_correlation(y, feature_name):
   x = features[feature_name]
   # some preprocessing of x and y for NAs
  ....
   corr, _ = pearsonr(x, y)
   return corr
```

And then you'd be able to compute the whole column and store it in the dest table in a single shot:
```
ensemble[f'{feature_col}_correlation']] = [ calc_correlation_with_na(targets[target_variable], feature_name ) for feature_name in row[feature_col] ] 
```